### PR TITLE
added eors

### PR DESCRIPTION
**Pull request by prakharcode**
@giordano any lights on this error? 
```
julia> a = rand(3,3)
3×3 Array{Float64,2}:
 0.286184  0.497562  0.0348385
 0.233051  0.532657  0.925174 
 0.301579  0.295383  0.814692 

julia> equations_of_origins(a, 2)
2.165203821361782

julia> ERFA.eors(a, 2)
0.9459823155896738
```

### CODE REVIEW

**Comment by giordano** [comment]
There are two issues:

1. `÷` in Julia performs integer division, simply leave `/` for usual division, just like in the C code
2. you should know by now what's the main issue :wink:

**Comment by prakharcode** [comment]
1. I tried both / and ÷
2. Okay, let me have a look again

**Comment by giordano** [comment]
In Julia, `÷` and `/` do very different things.

In C there isn't a native operator to perform integer division between floating point numbers.  Instead, division between integer types is integer division, but that's not the case here  (all operands in the `eors` code are `double`).

File: src/AstroBase.jl [hunk 1]
```diff
@@ -115,4 +115,25 @@ function tio_locator(jd1, jd2)
     -47e-6 * t * sec2rad(1)
 end
 
+"""
+    equations_of_origins(rnpb, s)
```
Comment (helgee) [inline src/AstroBase.jl]: `equation_of_origins`

**Comment by helgee** [comment]
@prakharcode This is a row-major vs. column-major matrix issue. You will see that if you pass the transpose of your matrix to the function the result will be correct.

**Comment by helgee** [comment]
@prakharcode BTW, you could also port the test cases from ERFA so you do not have to come up with your own. See here: https://github.com/liberfa/erfa/blob/master/src/t_erfa_c.c

**Comment by helgee** [comment]
@giordano Now I ruined your lesson because I did not read your reply closely 🙈 Sorry 😜

File: src/AstroBase.jl [hunk 2]
```diff
@@ -481,8 +482,32 @@ function xy06(jd1, jd2)
...
+"""
+function equation_of_origins(rnpb, s)
+    x = rnpb[1, 3]
+    ax = x / (1.0 + rnpb[3, 3])
+    xs = 1.0 - (ax * x)
+    ys = -ax * rnpb[2, 3]
+    zs = -x
+    p = (rnpb[1, 1] * xs) + (rnpb[2, 1] * ys) + (rnpb[3, 1] * zs)
+    q = (rnpb[1, 2] * xs) + (rnpb[2, 2] * ys) + (rnpb[3, 2] * zs)
+    ((p != 0) || (q != 0)) ? s - atan2(q, p) : s
```
Comment (helgee) [inline src/AstroBase.jl]: Parens are unnecessary.

File: src/AstroBase.jl [hunk 3]
```diff
@@ -481,8 +482,32 @@ function xy06(jd1, jd2)
         end
         ialast = ia - 1
     end
-
     sec2rad((xpr + (xyls[1] + xypl[1]) / 1e6)), sec2rad(ypr + (xyls[2] + xypl[2]) / 1e6)
 end
 
+"""
+    equations_of_origins(rnpb, s)
```
Comment (helgee) [inline src/AstroBase.jl]: equation_of_origins

File: src/AstroBase.jl (hunk 2 above)
Comment (helgee) [reply src/AstroBase.jl]: `atan(q, p)` see https://github.com/JuliaLang/julia/issues/27248

File: src/AstroBase.jl [hunk 4]
```diff
@@ -481,8 +482,32 @@ function xy06(jd1, jd2)
...
+# Example
+
+```jldoctest
+
+```
+"""
+function equation_of_origins(rnpb, s)
+    x = rnpb[1, 3]
+    ax = x / (1.0 + rnpb[3, 3])
+    xs = 1.0 - (ax * x)
```
Comment (helgee) [inline src/AstroBase.jl]: Actually, all parens beyond this point should be removed (except for the function call) 😂 https://en.wikipedia.org/wiki/Order_of_operations#Mnemonics 😜

**Comment by helgee** [comment]
There were some unrelated changes (deleted lines etc.) but I have fixed them myself.

**Pull request merged**